src/main/scala-2.13+/scala/concurrent/stm/compat/CompatTMap.scala

```diff
@@ -0,0 +1,35 @@
+package scala.concurrent.stm
+package compat
+
+import scala.collection._
+import scala.collection.{mutable, immutable}
+import scala.language.higherKinds
+
+private[stm] trait TMapViewCompanion extends MapFactory[TMap.View] {
+  def from[A, B](it: IterableOnce[(A, B)]): TMap.View[A, B] = {
+    val b = newBuilder[A, B]
+    if (it.knownSize >= 0)
+      b.sizeHint(it.knownSize)
+
+    b ++= it
+    b.result()
+  }
```

Comment:
I think we should inline the body of the `TMap.apply` method here, and then remove the `TMap.apply` override.

Reply:
What does TMap.apply overrides?

Reply:
It overrides `MapFactory.apply`, which simply calls `from`.

Reply:
We can do this, however, we don't have knownSize on `TraversableOnce` in 2.12

```scala
type IterableOnce[+X] = scala.collection.TraversableOnce[X]
```